Islandora-CLAW/controlled_access_terms
# TypedRelation override JSON-LD relationship predicates

**GitHub Issue**: [Issue 889](https://github.com/Islandora-CLAW/CLAW/issues/889)

# What does this Pull Request do?

This PR implements the JSON-LD `hook_jsonld_alter_normalized_array()` to allow ['typed_relation' fields](https://github.com/Islandora-CLAW/controlled_access_terms/blob/8.x-1.x/src/Plugin/Field/FieldType/TypedRelation.php) to add relationships based on the field's selected predicate. It also adds the field to some of the default configuration taxonomy terms with some schema.org default predicates and removes the generic relationship entity reference fields.

# What's new?

* Changes default taxonomy terms feature to such that they use typed_relation instead of a generic entity reference field and, if used, populates the JSON-LD for the term based on the selected predicate.
* Added controlled_access_terms_jsonld_alter_normalized_array() to update the JSON-LD normalized array with the field's predicate and value.
* Does this change require documentation to be updated? No (because we didn't have any to begin with; we should change that...)
* Does this change add any new dependencies? A soft dependency on jsonld (the module still works without it, this code is only called if jsonld is enabled).
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? No
* Could this change impact execution of existing code? No. 

# How should this be tested?

- Apply the PR
- Import the Feature changes
- Create some terms with relationships to other terms using one of the typed relation fields
- View the term's jsonld (?_format=jsonld) to see the predicate/value you added.

# Additional Notes:
This can be tested on a vanilla Drupal with controlled_access_terms enabled, but you will also need to enable the jsonld module, which is not listed as a dependency, to verify it works.

# Interested parties
@Islandora-CLAW/committers
